zk_toolbox/crates/zk_inception/src/commands/chain/init.rs

```diff
@@ -54,15 +57,29 @@ pub async fn init(
     ecosystem_config: &EcosystemConfig,
     chain_config: &ChainConfig,
 ) -> anyhow::Result<()> {
+    let mut ecosystem_ports = EcosystemPortsScanner::scan(shell)?;
     copy_configs(shell, &ecosystem_config.link_to_code, &chain_config.configs)?;
 
+    if !init_args.no_port_reallocation {
+        ecosystem_ports.allocate_ports_in_yaml(
+            shell,
+            &chain_config.path_to_general_config(),
+            chain_config.id,
+        )?;
+    }
     let mut general_config = chain_config.get_general_config()?;
-    apply_port_offset(init_args.port_offset, &mut general_config)?;
-    let ports = ports_config(&general_config).context(MSG_PORTS_CONFIG_ERR)?;
+    let offset = ((chain_config.id - 1) * 100) as u16;
+    let consensus_port_range = DEFAULT_CONSENSUS_PORT + offset..PORT_RANGE_END;
+    let consensus_port =
```

Comment:
can't we reuse the ecosystem ports for it?

Reply:
You mean adding a function to `EcosystemPorts` to allocate consensus port?

Reply:
Another approach is to add `ConsensusConfig` to file_based/general.yaml
including the following fields:
* server_addr
* public_addr
* max_payload_size
* max_batch_size
* gossip_dynamic_inbound_limit

Then we need a special case for `server_addr` and `public_addr` in the function `traverse_allocate_ports_in_yaml`